web/src/lib/components/shared-components/focus-trap.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+  import { shortcuts } from '$lib/utils/shortcut';
+  import { onMount, onDestroy } from 'svelte';
+
+  let container: HTMLElement;
+  let triggerElement: HTMLElement;
+
+  onMount(() => {
+    triggerElement = document.activeElement as HTMLElement;
+    const focusableElements = getFocusableElements();
+    focusableElements[0]?.focus();
+  });
+
+  onDestroy(() => {
+    triggerElement?.focus();
+  });
+
+  const getFocusableElements = () => {
+    return Array.from(
```

Comment:
For reference, another example implementation for finding focusable elements is the `tabbable` package.

https://github.com/focus-trap/tabbable